src/main/java/frc/robot/commands/shooter/SemiAutoShooterAssembly.java

```diff
@@ -0,0 +1,190 @@
+package frc.robot.commands.shooter;
+
+import edu.wpi.first.wpilibj2.command.CommandScheduler;
+import frc.robot.commands.drivetrain.EncoderDrive;
+import frc.robot.commands.drivetrain.TurnByEncoder;
+import frc.robot.devices.Lemonlight;
+import frc.robot.oi.inputs.OIButton;
+import frc.robot.subsystems.Conveyor;
+import frc.robot.subsystems.Conveyor.ConveyorState;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.Shooter;
+import frc.robot.utilities.Functions;
+import frc.robot.utilities.lists.AxisPriorities;
+
+
+/**
+ * Command for running the shooter in semi auto mode.
+ */
+public class SemiAutoShooterAssembly extends FullAutoShooterAssembly {
+
+    // subsystems
+    private Shooter shooter;
+    private Conveyor conveyor;
+    private Drivetrain drivetrain;
+
+    // OI
+    OIButton shootButton;
+    OIButton.PrioritizedButton prioritizedShootButton;
+
+    // tracker variables
+    private ConveyorState indexState;
+    private boolean isBallIndexed;
+    private double motorSpeed;
+    private double limelightDistanceEstimate;
+    private boolean limelightHasTarget;
+    private ConveyorState teamColor;
+    private double smoothedHorizontalOffset;
+    private double targetMotorSpeed;
+
+    // constants
+    private static final double
+        TURN_DEGREES_PER_CYCLE = 1.0,
+        MOVE_FORWARD_PER_CYCLE = 1.0,
+        SHOOTER_RANGE = 10.0,
+        HOOD_UP_RANGE = 5,
+        TARGET_HORIZONTAL_ACCURACY = 3,
+        TARGET_WRONG_COLOR_MISS = 45,
+        TARGET_MOTOR_SPEED_ACCURACY = 3;
+
+
+    // devices
+    private Lemonlight limelight;
+    
+    /**
+     * Command for running the shooter in semi auto mode.
+     *
+     * @param shooter The shooter subsystem.
+     * @param conveyor The conveyor subsystem.
+     * @param drivetrain The drivetrain subsystem.
+     * @param limelight The limelight device.
+     * @param shootButton The button to enable the shooter.
+     */
+    public SemiAutoShooterAssembly(Shooter shooter,
+        Conveyor conveyor,
+        Drivetrain drivetrain,
+        Lemonlight limelight,
+        OIButton shootButton) {
+        super(shooter, conveyor, drivetrain, limelight);
+        this.shooter = shooter;
+        this.conveyor = conveyor;
+        this.shootButton = shootButton;
+
+        addRequirements(shooter, conveyor);
+    }
+
+    @Override
+    public void initialize() {
+        shooter.stop();
+
+        prioritizedShootButton = shootButton.prioritize(AxisPriorities.DEFAULT);
+        
+    }
+
+    @Override
+    public void execute() {
```

Comment:
you should not re-impliment the whole excuite loop, instead break the full auto exciute loop into many functon calls and then override the ones that change in this file